sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -219,6 +219,10 @@ protected String inputToSQL(Expression input) {
     }
   }
 
+  protected String inputToCaseWhenSQL(Expression input) {
```

Comment:
This is very specific to `MsSqlServerSQLBuilder`, can we put it there?

Reply:
actually there are more databases that have this issue. perhaps in the future someone would need such a fix

Reply:
then let's make the method name clear about it, e.g. `convertBooleanToInt`

Reply:
but then it conflicts with the if (always true)
I think we should keep everything as is (function converts input to CASE WHEN regardless, it is up to devs to decide when to use it. it is just too much responsibility to this utility function)

Reply:
Lets add it in JdbcDialect

Reply:
done.

Reply:
I've split it to two functions